pkg/auth/serviceaccount_test.go

```diff
@@ -211,3 +211,31 @@ func TestSetupOIDCServiceAccount(t *testing.T) {
 		t.Errorf("SetupOIDCServiceAccount didn't set TriggerConditionOIDCIdentityCreated Status")
 	}
```

Comment:
Should I add the enabled, then disabled case in `TestSetupOIDCServiceAccount` with - 
```
	// authentication-oidc feature disabled, after enabling so service account was created.
	listers = rttestingv1.NewListers([]runtime.Object{expected})
	err = SetupOIDCServiceAccount(ctx, feature.Flags{
		feature.OIDCAuthentication: feature.Disabled,
	}, listers.GetServiceAccountLister(), kubeclient.Get(ctx), gvk, objectMeta, &trigger.Status, func(as *duckv1.AuthStatus) {
		trigger.Status.Auth = as
	})

	if err != nil {
		t.Errorf("SetupOIDCServiceAccount  failed: %s", err)
	}

	if trigger.Status.Auth != nil {
		t.Errorf("SetupOIDCServiceAccount setAuthStatus  failed")
	}

	// match OIDCIdentityCreated condition
	matched = false
	for _, condition := range trigger.Status.Conditions {
		if condition.Type == eventingv1.TriggerConditionOIDCIdentityCreated {
			if condition.Reason == "authentication-oidc feature disabled" {
				matched = true
			}
		}
	}

	if !matched {
		t.Errorf("SetupOIDCServiceAccount didn't set TriggerConditionOIDCIdentityCreated Status")
	}
```

used `expected `in `listers`, since the service account is already created by `EnsureOIDCServiceAccountExistsForResource` when setup function is called first time  but it's not updated in listers, and when we try to create the service account, it throws an error, that it already exists (we can create with another name though).
Should I add the above code, or the `TestDeleteOIDCServiceAccountIfExists` is enough?

Reply:
Makes sense IMO, as we updated the logic in `SetupOIDCServiceAccount`